docs/source/tutorials/Microlensing.ipynb

```diff
@@ -0,0 +1,318 @@
+   "metadata": {},
+   "source": [
+    "Now we can set up the parameters for our simulation. We can look at the order of the parameters expected by our simulator by looking at `x_order`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim.x_order"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Microlensing Physical Parameters\n",
+    "th_ein = 0.3  # Einstein radius in arcsec\n",
+    "\n",
+    "# Microlensing Model Parameters\n",
+    "t0 = 0.2  # time at peak magnification\n",
+    "u0 = 0.5  # minimum impact parameter (u(t=t_0)) in units of th_ein\n",
+    "tE = 0.05  # Einstein timescale\n",
+    "# rho = 3.0  # source size in units of lens Einstein radii\n",
+    "rho = 2.0  # source size in units of lens Einstein radii\n",
+    "\n",
+    "gamma = 0.0  # linear limb darkening coefficient"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "B = 64  # Batch size\n",
+    "ts = torch.linspace(-6 * tE + t0, 6 * tE + t0, B).view(-1, 1)  # Create time grid\n",
+    "\n",
+    "# Calculate source position at each time in arcsec\n",
+    "x0s = (ts - t0) / (tE) * th_ein  # Shape is [B, 1]\n",
+    "\n",
+    "# Since source is assumed radially symmetric, can set all motion in the x direction\n",
+    "y0s = -u0 * th_ein * torch.ones_like(x0s)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x = torch.tensor(\n",
+    "    [\n",
+    "        0.0,  # z_l\n",
+    "        0.0,  # x0\n",
+    "        0.0,  # y0\n",
+    "        th_ein,  # th_ein\n",
+    "        0.0,  # x0\n",
+    "        0.0,  # y0\n",
+    "        th_ein * rho,  # theta_s\n",
+    "        5.0,  # Ie,\n",
+    "        gamma,  # gamma\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "# Create the input tensor for the simulator. In this simple example, everything is static except for the lens position\n",
+    "newx = x.repeat(B, 1)\n",
+    "newx[:, 1] = x0s.view(-1)\n",
+    "newx[:, 2] = y0s.view(-1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running the simulation and visualzing the output\n",
+    "\n",
+    "Running a batched simulation over different parameters (in our case, these correspond to the lens motion in discrete time-steps by construction) is now as easy as using `vmap` on our simulator. We can then visualize the total output"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "images, magnifications = torch.vmap(sim)(newx)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create animation\n",
```

Comment:
You can add a tag to this cell `hide-input` so that users will only see the plot and not all the plt commands, it makes for a cleaner look.